File: 알고리즘/CodingTest/src/Retry/Caesar.java
package Retry;

// https://programmers.co.kr/learn/courses/30/lessons/12926

public class Caesar {
    public String solution(String s, int n) {
        String answer = "";

        for(int i = 0; i < s.length(); i++) {
            char a = s.charAt(i);

            if(a >= 'a' && a <= 'z') {
                if(a + n > 'z') answer += (char)(a + n - 26);
                else answer += (char)(a + n);
            } else if (a >= 'A' && a <= 'Z') {
                if(a + n > 'Z') answer += (char)(a + n - 26);
                else answer += (char)(a + n);
            } else answer += (char)(a); // 공백 처리
        }

        return answer;
    }
}

File: Algorithm/Algorithm_py/programmers/programmers1/Caesar.py
# https://programmers.co.kr/learn/courses/30/lessons/12926?language=python3
def solution(s, n):
  answer = ''
  
  for i in s:
    if 'a' <= i <= 'z':
      answer += chr((ord(i) - ord('a') + n) % 26 + ord('a'))
    elif 'A' <= i <= 'Z':
      answer += chr((ord(i) - ord('A') + n) % 26 + ord('A'))
    else:
      answer += i
  
  return answer